Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,315 @@ public static class BalanceStats
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static bool IsDecArrayNullOrEmpty(decimal[] array)
+        {
+            if (array == null || array.Length == 0)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static decimal[] GetBalances(string[] balance)
+        {
+            if (IsStrArrayNullOrEmpty(balance))
+            {
+                return null;
+            }
+            decimal[] auxArray = new decimal[balance.Length];
+
+            int j = 0;
+            
+            for (int i = 0; i < balance.Length; i++)
+            {
+                if (!string.IsNullOrWhiteSpace(balance[i]))
+                {
+                    decimal.TryParse(balance[i], out auxArray[j]);
+                    j++;
+                }
+                
+            }
+
+            return auxArray[..j];
+        }
+
+
+
+        private static decimal MaxBalance(decimal[] array)
+        {
+            if (IsDecArrayNullOrEmpty(array))
+            {
+                return decimal.MinValue;
+            }
+
+            decimal max = decimal.MinValue;
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (max < array[i])
+                {
+                    max = array[i];
+                }
+            }
+            return max;
+        }
+
+        private static decimal MinBalance(decimal[] array)
+        {
+            if (IsDecArrayNullOrEmpty(array))
+            {
+                return decimal.MinValue;
+            }
+
+            decimal min = decimal.MaxValue;
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (min > array[i])
+                {
+                    min = array[i];
+                }
+            }
+            return min;
+        }
+
+
+        private static decimal[] GetDiff(decimal[] array)
+        {
+            if (IsDecArrayNullOrEmpty(array))
+            {
+                return array;
+            }
+
+            if (array.Length == 1)
+            {
+                return new decimal[] { 0 };
+            }
+
+            decimal[] auxArray = new decimal[array.Length - 1];
+
+            for (int i = 0; i < array.Length - 1; i++)
+            {
+                auxArray[i] = array[i+1] - array[i];
+            }
+
+            return auxArray;
+        }
+
+        /// <summary>
+        /// Replace the last comma with the string " and" in a string with values separated by comma.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns>Return a string with the last "," replaced by " and"</returns>
+        private static string ReplaceLastCommaWithAnd(string value) 
+        {
+            int lastComma = value.LastIndexOf(',');
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'lastComma'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5069037)

Reply:
I don't understand this Kai, I'm using the variable in the next sentence:
```
int lastComma = value.LastIndexOf(',');
return value[..lastComma] + " and" + value[(lastComma + 1)..];
```

Reply:
Ignore it, it was a poor analysis of a bot.

Reply:
Although, I think I'll use the stringbuilder to do the same thing and delete the function, but still I'll use the variable to calculate the last comma position. I think it is more readable. For example:
```
StringBuilder b = new StringBuilder("uno, dos, tres y mas texto");
int pos = b.ToString().LastIndexOf(",");
b.Replace(",", " y", pos, 1); // b = "uno, dos y tres y mas texto"
```